runtime/src/main/java/org/corfudb/runtime/collections/ISMRMap.java

```diff
@@ -133,8 +133,28 @@
      */
     @MutatorAccessor(name = "put", undoFunction = "undoPut", undoRecordFunction = "undoPutRecord")
     @Override
-    V put(K key, V value);
+    V put(@ConflictParameter K key, V value);
 
+
+    /**
+     * Associates the specified value with the specified key in this map
+     * without return the previous value.  If the map previously contained a
+     * mapping for the key, the old value is replaced by the specified value.
+     * (A map <tt>m</tt> is said to contain a mapping for a key <tt>k</tt> if
+     * and only if {@link #containsKey(Object) m.containsKey(k)} would return
+     * <tt>true</tt>.)
+     *
+     * @param key   key with which the specified value is to be associated
+     * @param value value to be associated with the specified key
+     * @throws UnsupportedOperationException if the <tt>put</tt> operation
+     *                                       is not supported by this map
+     * @throws ClassCastException            if the class of the specified key or value
+     *                                       prevents it from being stored in this map
+     * @throws NullPointerException          if the specified key or value is null
+     *                                       and this map does not permit null keys or values
+     * @throws IllegalArgumentException      if some property of the specified key
+     *                                       or value prevents it from being stored in this map
+     */
```

Comment:
First, I believe that key is a (write-) conflict parameter for a blindPut, no?

Second, how do we handle conflict resolution between two transaction on the same object, one of which has specific conflict parameters, the other has no explicit ones?